src/ExtraThemeAndPluginHeaders.php

```diff
@@ -0,0 +1,69 @@
+<?php
+/**
+ * Class ExtraThemeAndPluginHeaders.
+ *
+ * @package AmpProject\AmpWP
+ */
+
+namespace AmpProject\AmpWP;
+
+use AmpProject\AmpWP\Infrastructure\Registerable;
+use AmpProject\AmpWP\Infrastructure\Service;
+
+/**
+ * Registers the 'AMP' extra header for themes and plugins.
+ *
+ * @package AmpProject\AmpWP
+ * @since 2.0
+ * @internal
+ */
+final class ExtraThemeAndPluginHeaders implements Service, Registerable {
+
+	/**
+	 * Header name.
+	 *
+	 * @var string
+	 */
+	const AMP_HEADER = 'AMP';
+
+	/**
+	 * Values which indicate the theme has 'amp' theme support.
+	 *
+	 * @var string[]
+	 */
+	const AMP_THEME_SUPPORT_VALUES = [
+		'theme-support',
+		'theme_support',
+		'true',
+		'yes',
+	];
+
+	/**
+	 * Register the service with the system.
+	 *
+	 * @return void
+	 */
+	public function register() {
+		// Filter must be added as soon as possible since once wp_get_themes() is called, the results are cached.
+		add_filter( 'extra_theme_headers', [ $this, 'filter_extra_headers' ] );
+
+		// Note that get_plugins() may have already been called, resulting in this returning stale data. See \WC_Helper::get_local_woo_plugins().
+		add_filter( 'extra_plugin_headers', [ $this, 'filter_extra_headers' ] );
```

Comment:
I don't need to know now, but I'm curious what you have in mind for the plugin header. It doesn't seem to be used anywhere now, but I can imagine a few possible uses down the line.

Reply:
I can probably just remove it for now. The idea is that plugins which indicate they are AMP-compatible could have some badge or something indicating they shouldn't need to be evaluated for AMP-compatibility when activating the AMP plugin.

Reply:
Yeah, it might be nice to work this into the site scan flow in some way but I'm not sure exactly how. If you do remove it, it's easily restored.

Reply:
Removed in 4084b71862577683fa9dc39eca8895df5bbf4868